pinot-core/src/main/java/org/apache/pinot/core/periodictask/PeriodicTaskScheduler.java

```diff
@@ -108,4 +111,59 @@ public synchronized void stop() {
       _tasksWithValidInterval.parallelStream().forEach(PeriodicTask::stop);
     }
   }
+
+  /** @return true if task with given name exists; otherwise, false. */
+  public boolean hasTask(String periodicTaskName) {
+    for (PeriodicTask task : _tasksWithValidInterval) {
+      if (task.getTaskName().equals(periodicTaskName)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  /** @return List of tasks name that will run periodically. */
+  public List<String> getTaskNameList() {
+    List<String> taskNameList = new ArrayList<>();
+    for (PeriodicTask task : _tasksWithValidInterval) {
+      taskNameList.add(task.getTaskName());
+    }
+
+    return taskNameList;
+  }
+
+  private PeriodicTask getPeriodicTask(String periodicTaskName) {
+    for (PeriodicTask task : _tasksWithValidInterval) {
+      if (task.getTaskName().equals(periodicTaskName)) {
+        return task;
+      }
+    }
+    return null;
+  }
+
+  /** Execute {@link PeriodicTask} immediately on the specified table. */
+  public void scheduleNow(String periodicTaskName, @Nullable Properties periodicTaskProperties) {
+    // Each controller may have a slightly different list of periodic tasks if we add, remove, or rename periodic
+    // task. To avoid this situation, we check again (besides the check at controller API level) whether the
+    // periodic task exists.
+    PeriodicTask periodicTask = getPeriodicTask(periodicTaskName);
+    if (periodicTask == null) {
+      throw new IllegalArgumentException("Unknown Periodic Task " + periodicTaskName);
+    }
+
+    LOGGER.info("Immediately executing periodic task {}", periodicTaskName);
```

Comment:
This is misleading if the task ends up waiting on a lock, and that could be a very long time. You may want to say "scheduling the task to run"

Reply:
Modified text to clarify.